crates/bevy_window/src/window.rs

```diff
@@ -511,6 +518,22 @@ impl Window {
     pub fn is_focused(&self) -> bool {
         self.focused
     }
+
+    #[inline]
+    pub fn icon(&self) -> Option<WindowIcon> {
+        self.icon.clone()
+    }
```

Comment:
```suggestion
    pub fn icon(&self) -> Option<&WindowIcon> {
        self.icon.as_ref()
    }
```
There's not need for us to clone the icon. 
We can just return a reference to it and then have the called clone this if the want to.

Reply:
I still very much depend on the rust compiler to tell what works and does not.
This seems like a no-brainer though,
Thanks,